sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/AlreadyExistException.scala

```diff
@@ -61,6 +61,11 @@ class TableAlreadyExistsException(errorClass: String, messageParameters: Map[Str
     this(errorClass = "TABLE_OR_VIEW_ALREADY_EXISTS",
       messageParameters = Map("relationName" -> quoteNameParts(table)))
   }
+
+  def this(tableIdent: Identifier) = {
```

Comment:
Here is the definition used in 3.3.

```
class TableAlreadyExistsException(message: String, cause: Option[Throwable] = None)
  extends AnalysisException(message, cause = cause) {
  def this(db: String, table: String) = {
    this(s"Table or view '$table' already exists in database '$db'")
  }

  def this(tableIdent: Identifier) = {
    this(s"Table ${tableIdent.quoted} already exists")
  }
}
```

Technically, there was another constructor accepting `message`. Shall we restore it too?

Reply:
That would require adding back error templates like `_LEGACY_ERROR_TEMP_1115`.

Reply:
We either need to add back old error templates or just directly construct `AnalysisException` with a message. Same is also need to restore old constructors for `NoSuchIndexException`, `IndexAlreadyExistsException`.

Reply:
I am inclined to add back all constructors but I'll wait to hear more from others.

Reply:
@aokolnychyi +1 for adding back all constructors. 
Let's do it and merge this one today.

Reply:
Sounds good, will add them now.